tests/run/t6637.scala

```diff
@@ -1,8 +0,0 @@
-
-object Test extends dotty.runtime.LegacyApp {
```

Comment:
In this test `List()` is evaluated to `List[Nothing]` which leads to `head.isInstanceOf[A with B]` will always be true since `Nothing` is a subtype of `A`.

The new mini-phase will report an error here, as such the test will fail because it won't compile and `println("ok")` will never run.

Of course, the mini-phase could just warn here...decide what you think is best and I'll adjust the PR.